cmd/notation/sign.go

```diff
@@ -189,5 +215,23 @@ func prepareSigningOpts(opts *signOpts) (notation.SignOptions, error) {
 		},
 		UserMetadata: userMetadata,
 	}
+	if opts.tsaServerURL != "" {
+		// timestamping
+		fmt.Printf("Timestamping with TSA %q\n", opts.tsaServerURL)
+		signOpts.Timestamper, err = tspclient.NewHTTPTimestamper(&http.Client{Timeout: TimestampingTimeout}, opts.tsaServerURL)
+		if err != nil {
+			return notation.SignOptions{}, fmt.Errorf("cannot get http timestamper for timestamping: %v", err)
+		}
+
+		rootCerts, err := corex509.ReadCertificateFile(opts.tsaRootCertificatePath)
+		if err != nil {
+			return notation.SignOptions{}, err
+		}
+		if len(rootCerts) == 0 {
+			return notation.SignOptions{}, fmt.Errorf("cannot read tsa root certificate from %q", opts.tsaRootCertificatePath)
```

Comment:
Should we mention that the file is empty?

Reply:
If the file contains more than one certs, should we remind the user that we will only use the first one.

Reply:
The file is not necessarily empty though. It's just there's no valid x509 certificate in it. The root cert downloaded from TSA's webpage is usually a single cert file.

Reply:
Yes, we need to mention the reason and expected format to help the user figure out the issue.

> cannot read tsa root certificate from %q

It doesn't explain the reason.

Reply:
Updated.